### PR TITLE
ccache preliminary support

### PR DESCRIPTION
**Pull request by jmlapre**
@trilinos/framework 

## Motivation
Trilinos is a massive codebase often building for hours at a time.  However, changes are often incremental and fairly redundant.  Adding [ccache](https://ccache.dev/) would drastically reduce the amount of time required per-build. This PR adds some minimal supporting changes and somehow fell off our internal boards unmerged.

### CODE REVIEW

**Comment by sebrowne** [comment]
Adding this to the configure command does effectively the same thing:
`-DCMAKE_C_COMPILER_LAUNCHER=ccache
-DCMAKE_CXX_COMPILER_LAUNCHER=ccache`

Is there  a reason to do it with wrappers instead?

**Comment by jmlapre** [comment]
They add support for Xcode and also define `CCACHE_CPP2`.  But if we are OK with cutting Xcode then we could simply define those variables as well as `CCACHE_CPP2` and call it a day.

File: CMakeLists.txt [hunk 1]
```diff
@@ -69,6 +69,35 @@ INCLUDE(${CMAKE_CURRENT_LIST_DIR}/ProjectName.cmake)
 # not in an include file :-(
 PROJECT(${PROJECT_NAME} NONE)
 
+if("${CCACHE_NODISABLE}")
```
Comment (sebrowne) [inline CMakeLists.txt:72]: I think this is more descriptive as `$ENV{CCACHE_NODISABLE}`

File: CMakeLists.txt [hunk 2]
```diff
@@ -69,6 +69,35 @@ INCLUDE(${CMAKE_CURRENT_LIST_DIR}/ProjectName.cmake)
...
+        set(CMAKE_XCODE_ATTRIBUTE_CC         "${CMAKE_BINARY_DIR}/launch-c")
+        set(CMAKE_XCODE_ATTRIBUTE_CXX        "${CMAKE_BINARY_DIR}/launch-cxx")
+        set(CMAKE_XCODE_ATTRIBUTE_LD         "${CMAKE_BINARY_DIR}/launch-c")
+        set(CMAKE_XCODE_ATTRIBUTE_LDPLUSPLUS "${CMAKE_BINARY_DIR}/launch-cxx")
+    else()
+        # Support Unix Makefiles and Ninja
+        set(CMAKE_C_COMPILER_LAUNCHER   "${CMAKE_BINARY_DIR}/launch-c")
+        set(CMAKE_CXX_COMPILER_LAUNCHER "${CMAKE_BINARY_DIR}/launch-cxx")
+    endif()
+ endif()
```
Comment (sebrowne) [inline CMakeLists.txt:98]: If ccache was requested but not found, I'd like to get a message about it

**Review by sebrowne — CHANGES_REQUESTED** [review]

**Review by sebrowne — APPROVED** [review]

**Pull request merged**